guava/src/com/google/common/base/Ascii.java

```diff
@@ -629,4 +630,155 @@ private static int getAlphaIndex(char c) {
     // Fold upper-case ASCII to lower-case and make zero-indexed and unsigned (by casting to char).
     return (char) ((c | 0x20) - 'a');
   }
+
+  /**
+   * Searches through {@code source} to find {@code target}, ignoring the case of
+   * any ASCII alphabetic characters between {@code 'a'} and {@code 'z'}
+   * or {@code 'A'} and {@code 'Z'} inclusive.
+   *
+   * @param source       the characters being searched.
+   * @param sourceOffset offset of the source string.
+   * @param sourceCount  count of the source string.
+   * @param target       the characters being searched for.
+   * @param targetOffset offset of the target string.
+   * @param targetCount  count of the target string.
+   * @param fromIndex    the index to begin searching from.
+   */
+  private static int indexOfIgnoreCase(char[] source, int sourceOffset, int sourceCount,
+                                       char[] target, int targetOffset, int targetCount,
+                                       int fromIndex) {
+    if (fromIndex >= sourceCount) {
+      return (targetCount == 0 ? sourceCount : -1);
+    }
+    if (fromIndex < 0) {
+      fromIndex = 0;
+    }
+    if (targetCount == 0) {
+      return fromIndex;
+    }
+
+    char first = toLowerCase(target[targetOffset]);
+    int max = sourceOffset + (sourceCount - targetCount);
+
+    for (int i = sourceOffset + fromIndex; i <= max; i++) {
+      /* Look for first character. */
+      if (toLowerCase(source[i]) != first) {
+        while (++i <= max && toLowerCase(source[i]) != first) {
+          ;
+        }
+      }
+
+      /* Found first character, now look at the rest of v2 */
+      if (i <= max) {
+        int j = i + 1;
+        int end = j + targetCount - 1;
+        for (int k = targetOffset + 1; j < end && toLowerCase(source[j])
+            == toLowerCase(target[k]); j++, k++) {
+          ;
+        }
+
+        if (j == end) {
+          /* Found whole string. */
+          return i - sourceOffset;
+        }
+      }
+    }
+    return -1;
+  }
+
+  /**
+   * Returns the index within the {@code string} of the first occurrence of the {@code subString},
+   * starting at {@code fromIndex}, ignoring the case of any ASCII alphabetic characters
+   * between {@code 'a'} and {@code 'z'} or {@code 'A'} and {@code 'Z'} inclusive.
+   *
+   * @param string    the string to be searched in.
+   * @param subString the substring to search for.
+   * @param fromIndex the index from which to start the search.
+   * @return the index of the first occurrence of the {@code subString}, or {@code -1} if there is
+   *     no such occurrence.
+   *
+   * @since NEXT
+   */
+  public static int indexOfIgnoreCase(String string, String subString, int fromIndex) {
+    return indexOfIgnoreCase(string.toCharArray(), 0, string.length(),
```

Comment:
`toCharArray()` involves a linear time copy of array. It is slow, too.

Reply:
Thanks, I totally forgot about that. I fixed it now.